### ISSUE TITLE
redirectOutputToFile does not really work

### ISSUE THREAD

**Issue by stefanofornari** [beanshell/beanshell#516]
Hello,
if I am not mistaking, redirectOutputToFile()

https://github.com/beanshell/beanshell/blob/merge-fork-beanshell2/src/main/java/bsh/Interpreter.java#L1031-L1042

does not really do what aimed because it redirects the standard output only, while a parser (and therefore all commands) can use a provided output stream. The current behaviour can be achieved directly using System.setOut(out) by the caller. I would suggest at least to deprecate it.

**Comment by nickl-** [comment]
Perhaps this is legacy we should put it to vote.

**TODO: add poll.**

**Comment by nickl-** [comment]
Hmm this is really strange:

https://github.com/beanshell/beanshell/blob/60c74f11fec554dea1068d75cbbbcbe6ecbcb5f9/src/main/java/bsh/Interpreter.java#L1074-L1085

Why is it calling `System.setOut(out)` and `System.setErr(out)` when `Interpreter` has its own methods to redirect output.

https://github.com/beanshell/beanshell/blob/60c74f11fec554dea1068d75cbbbcbe6ecbcb5f9/src/main/java/bsh/Interpreter.java#L821-L829

**Comment by jimjag** [comment]
Agreed this was confusing. Added Interpreter.setOutputFile(String) in d589af31, a per-instance method that redirects this interpreter's own output/error streams (via its existing setOut()/setErr()) without touching System.out/System.err. The original static redirectOutputToFile(String) is now deprecated in favor of it, but kept as-is since it's still the mechanism used for the "-Doutfile" JVM startup property, which really is a JVM-wide redirect.
